cloudfront/wellcomelibrary.org/edge-lambda/src/paths.ts

```diff
@@ -0,0 +1,28 @@
+export type GetBNumberResult = string | Error;
+
+export function getBnumberFromPath(path: string): GetBNumberResult {
+  const splitPath = path.split('/');
+
+  // Match on paths like b1234567x / b12345678
+  const bNumberRegexp = /^[bB][0-9]{7}/;
+
+  if (splitPath[0] !== '') {
+    return Error(`Path ${path} does not start with /`);
```

Comment:
Big fan of flat guard clauses like this, by far the clearest way to represent this kind of logic 👍